test/array_ref.cpp

```diff
@@ -78,6 +78,23 @@ TEST(array_ref_copy) {
   ASSERT_EQ(scalar(), scalar_copy());
 }
 
+TEST(array_ref_incompatible_shape) {
```

Comment:
There's another test errors.cpp, though it's designed to fail at compile time, not runtime. Maybe we should add another test runtime_errors.cpp or something, and it should actually check for failure somehow. It would be hard to check for more than one though. Maybe re-define `assert` in that test somehow? That's super janky, but it could work.

Otherwise though, I'm fine with just not having even a commented out test for this change. It wasn't tested before either (when the failures were there, but opaque/hard to debug).

Reply:
Punting on adding more tests for now, but I kept the commented out lines. Otherwise let me know if you'd like to remove them, or for me to implement a runtime check.